### DISCUSSION TITLE
Does the govmomi API work on esxi free license?

### DISCUSSION THREAD

**Discussion by wouterdb** [vmware/govmomi#3027] [Community Help] [answered]
I currently have code that works against a licensed esxi, but will it also work against an esxi free license?

I've been able to find some redit posts that mention it won't, but on the vmware website, in the list of limitations, this is not mentioned. Does anyone know if it will work?

Thanks,
Wouter

**Comment by dougm** [comment, marked as answer]
I don't recall the details of ESXi free license API access, but thought it was read-only. Any limitation would be on the ESXi side, not govmomi/client side.
@karaatanassov may know of a vmware doc that explains?

**Reply by tenthirtyam** [reply]
API write access is not supported for use with a free VMware vSphere Hypervisor license.
